cfgov/unprocessed/js/modules/multi-select.js

```diff
@@ -0,0 +1,494 @@
+            self.placeholder = list[i].text;
+          }
+        }
+      }
+
+      this.filtered = this.optionData;
+    },
+
+    populateMarkup: function() {
+      // Create necessary elements
+      var self = this;
+
+      // Add a container for our markup and hide the default select elem
+      this.container = $.create( 'div', {
+        className: 'cf-multi-select',
+        around:    self.elem
+      } );
+
+      // Create all our markup but wait to manipulate the DOM just once
+      this.header = $.create( 'header', {
+        className: 'cf-multi-select_header'
+      } );
+
+      this.choices = $.create( 'ul', {
+        className: 'list__unstyled cf-multi-select_choices',
+        inside:    self.container
+      } );
+
+      this.search = $.create( 'input', {
+        className:   'cf-multi-select_search',
+        type:        'text',
+        placeholder: self.placeholder || 'Choose up to five',
+        inside:      self.header
+      } );
+
+      this.fieldset = $.create( 'fieldset', {
+        className:  'cf-multi-select_fieldset',
+        visibility: 'hidden'
+      } );
+
+      this.list = $.create( 'ul', {
+        className: 'list__unstyled cf-multi-select_options',
+        inside:    self.fieldset
+      } );
+
+      this.optionData.forEach( function( option ) {
+        var li = $.create( 'li', {
+          'data-option': option.value
+        } );
+
+        $.create( 'input', {
+          id:        option.value,
+          value:     option.value,
+          type:      'checkbox',
+          name:      self.name,
+          class:     'cf-input cf-multi-select_checkbox',
+          inside:    li
+        } );
+
+        $.create( 'label', {
+          for:       option.value,
+          innerText: option.text,
+          className: 'cf-multi-select_label',
+          inside:    li
+        } );
+
+        self.list.appendChild( li );
+      } );
+
+      // Write our new markup to the DOM
+      this.container.appendChild( self.header );
+      this.container.appendChild( self.fieldset );
+    },
+
+    get selected() {
+      return this.index > -1;
+    },
+
+    open: function() {
+      this.container.classList.add( 'active' );
+      this.fieldset.setAttribute( 'visibility', 'visible' );
+    },
+
+    close: function() {
+      this.container.classList.remove( 'active' );
+      this.fieldset.setAttribute( 'visibility', 'hidden' );
+      this.index = -1;
+    },
+
+    next: function() {
+      this.highlight( 'next' );
+    },
+
+    previous: function() {
+      this.highlight( 'prev' );
+    },
+
+    // Should not be used, highlights specific item without any checks!
+    highlight: function( direction ) {
+      var self = this;
+      var count = this.filtered.length;
+      var value;
+      var item;
+
+      if ( direction === 'next' && this.index < count - 1 ) {
+        this.index += 1;
+      } else if ( direction === 'prev' && this.index > -1 ) {
```

Comment:
How do you suggest? I'm also checking to make sure it's not the last item for `next` or the first item for `prev`, otherwise it would try to highlight something that doesn't exist